packages/webpack-config-flag-plugin/index.js

```diff
@@ -0,0 +1,100 @@
+const BasicEvaluatedExpression = require( 'webpack/lib/BasicEvaluatedExpression' );
+
+// Check that the given call expression is `config.isEnabled( 'flag' )` with
+// `config` as the default export or namespace, and return the `flag` literal value.
+const isCallOnDefaultOrNamespace = ( parser, moduleName, expr ) => {
+	return moduleName &&
+		parser.evaluate( moduleName ).identifier === 'imported var' &&
+		expr.callee.type === 'MemberExpression' &&
+		expr.callee.object.type === 'Identifier' &&
+		expr.callee.object.name === moduleName &&
+		expr.callee.property.type === 'Identifier' &&
+		expr.callee.property.name === 'isEnabled' &&
+		expr.arguments.length === 1 &&
+		expr.arguments[ 0 ].type === 'Literal'
+		? expr.arguments[ 0 ].value
+		: null;
+};
+
+// Check that the given call expression is `isEnabled( 'flag' )`
+// and return the `flag` literal value.
+const isNamedCall = ( parser, methodName, expr ) => {
+	return methodName &&
+		parser.evaluate( methodName ).identifier === 'imported var' &&
```

Comment:
This `parser.evaluate` call takes the currently walked AST node into account, and checks that the `config` variable refers to the imported binding and not some local scope override?

Reply:
I spent some time now learning about this `imported var` stuff. It works, but there is an advance warning: webpack 5 tracks the imports completely differently -- there's no longer any `imported var` rename.

Reply:
As far as I can tell from these undocumented APIs, `parser.evaluate` evaluates an expression in the current context. In this case, since we're giving it an identifier, the `identifier` property will be set, and will always be `'imported var'`. If there were something local overriding the import, this property would be set to something else (the name of the const, if I recall correctly).

We're building our castle on quicksand here, but such is life with `webpack` plugins 🤷‍♂

Reply:
> I spent some time now learning about this `imported var` stuff. It works, but there is an advance warning: webpack 5 tracks the imports completely differently -- there's no longer any `imported var` rename.

Oh, interesting! Any chance you could point me to that documentation? It's hard to decide what to build on with undocumented APIs...

Reply:
There's no documentation I'm aware of -- just the source.

Reply:
> the identifier property will be set, and will always be 'imported var'

That's because one of the parser plugins that walks through `import` and `export` statements will set `parser.renames.set( 'config', 'imported var' )` and `parser.evaluateExpression` will respect these renames. That's why you get `imported var` as the identifier name. Webpack 5 doesn't do that rename any more. I don't understand yet what's going on there.